include/tc/core/mapping_options.h

```diff
@@ -333,9 +230,14 @@ class SchedulerOptions : public SchedulerOptionsView {
 /// block sizes).  Provides static constructors for common operator options.
 /// Provides fluent (chainable) API for progressively modifying the options.
 class MappingOptions {
- private:
+ public:
+  virtual void concretize() = 0; // Make this class abstract
```

Comment:
Hmm... couldn't you just make the destructor abstact?
Actually, `toProtobufSerializedString` is already abstact.